crates/flux-middle/src/rty/expr.rs

```diff
@@ -1177,7 +1199,19 @@ mod pretty {
                     }
                 }
                 ExprKind::Aggregate(AggregateKind::Adt(def_id), flds) => {
-                    w!("{:?} {{ {:?} }}", def_id, join!(", ", flds))
+                    if let Some(genv) = cx.genv
+                        && let Ok(adt_sort_def) = genv.adt_sort_def_of(def_id)
+                    {
+                        let field_binds = adt_sort_def
+                            .field_names()
+                            .iter()
+                            .zip(flds)
+                            .map(|(name, value)| FieldBind { name: *name, value: value.clone() })
+                            .collect_vec();
+                        w!("{{ {:?} }}", join!(", ", field_binds))
```

Comment:
wait @ranjitjhala!. Didn't notice this, I'm not sure we should skip the `def_id` blindly. So, if you write 

```rust
S { v: v = S { x: 0, y: 0 } }
```

I think we do want to keep the `S` inside

Reply:
With this change we print `S {v: v = {x: 0, y: 0}}`

Reply:
Good catch, no worries, will fix.

Reply:
I'll elide the `S` (`DefId`) only when printing ... a top-level index?

Reply:
yes, that sounds good

Reply:
I used a newtype wrapper type trick to only elide the `DefId` for top-level indices, 

https://github.com/flux-rs/flux/pull/927/commits/a70354dcf924f1b9b96de10d6cbc0856dfed69b7

did you want to take a look? Or merge?